### PR TITLE
New time text input.

### PR DESCRIPTION
**Pull request by Luddelito**
image

The input will be formatted on blur.
image

### CODE REVIEW

File: src/util/time/__tests__/TimeTransformer.test.ts [hunk 1]
```diff
@@ -1,96 +1,156 @@
-import {
-  transformNumberTimeToString,
-  transformTimeStringToNumber,
-} from '../TimeTransformer';
+import { formattingTimeString, validUserInput } from '../TimeTransformer';
```
Comment (knegusen) [inline src/util/time/__tests__/TimeTransformer.test.ts]: Big like for all tests! Not sure if i like the test-names though...

File: src/util/time/TimeTransformer.ts [hunk 2]
```diff
@@ -1,70 +1,123 @@
-export const transformNumberTimeToString = (
-  time: number | undefined | null,
-): string | undefined => {
-  if (time == null) {
-    return undefined;
-  }
-  const timeStr = time.toString();
-  if (timeStr.length === 3) {
-    return `${timeStr.substr(0, 1)}:${timeStr.substr(1, 2)}`;
+export interface TimeFormattingResult {
```
Comment (knegusen) [inline src/util/time/TimeTransformer.ts]: Advanced one... =)

File: src/components/ui/form/date-time-input/TimeTextInput.tsx [hunk 3]
```diff
@@ -1,35 +1,64 @@
...
-  'TimeTextInput',
-)(
-  compose<DefaultTextInputProps, DefaultTextInputProps>(
-    withTheme,
-    withTimeProps,
-  )(DefaultTextInput),
-);
+  const theme = useTheme();
+
+  const onBlur = () => {
```
Comment (mattias800) [inline src/components/ui/form/date-time-input/TimeTextInput.tsx]: Use useCallback, don't create functions directly.

File: src/components/ui/form/date-time-input/TimeTextInput.tsx [hunk 4]
```diff
@@ -1,35 +1,64 @@
...
+    if (value) {
+      const formattedResult = formattingTimeString(value);
+      setValid(formattedResult.success);
+      if (formattedResult.success) {
+        onChange(formattedResult.time);
+      }
+    }
+  };
+
+  const updateValue = (time: string) => {
```
Comment (mattias800) [inline src/components/ui/form/date-time-input/TimeTextInput.tsx]: Use useCallback, don't create functions directly.

File: src/util/time/TimeTransformer.ts [hunk 5]
```diff
@@ -1,70 +1,123 @@
...
+  if (isNaN(m)) {
+    throw new Error('Minutes is not a number');
   }
 
-  const hours = parseInt(parts[0], 10);
-  const minutes = parseInt(parts[1], 10);
-
-  if (isNaN(hours)) {
-    throw new Error('Invalid time.');
+  if (m < 0 || m > 23) {
```
Comment (mattias800) [inline src/util/time/TimeTransformer.ts]: Minutes can probably go up to 59? :)

File: src/util/time/TimeTransformer.ts [hunk 6]
```diff
@@ -1,70 +1,123 @@
...
+      return minutes;
+    default:
+      throw new Error('Invalid minute string');
   }
-  if (hours < 0 || hours > 23) {
-    throw new Error('Invalid time.');
+  return minutes;
+};
+
+export const formattingTimeString = (time: string): TimeFormattingResult => {
```
Comment (mattias800) [inline src/util/time/TimeTransformer.ts]: Rename to `formatTimeString`.

File: src/util/time/__tests__/TimeTransformer.test.ts [hunk 7]
```diff
@@ -1,96 +1,156 @@
...
-    it('should throw if null', () => {
-      expect(() => transformTimeStringToNumber(null)).toThrow(
-        'Time is not set.',
-      );
-    });
-    it('should throw if undefined', () => {
-      expect(() => transformTimeStringToNumber(undefined)).toThrow(
-        'Time is not set.',
-      );
+  describe('formattingTimeString', () => {
```
Comment (mattias800) [inline src/util/time/__tests__/TimeTransformer.test.ts]: Just a reminder to rename to `formatTimeString` here as well :)

**Pull request merged**